extension.js

```diff
@@ -320,12 +348,8 @@ const NasaApodIndicator = new Lang.Class({
                     this._apiKeyErrorActions
                 );
             } else if (message.status_code == 429) {
-                if (verbose)
-                    Notifications.notifyError(_("Over rate limit (error 429)"),
-                        _("Get your API key at https://api.nasa.gov/ to have 1000 requests per hour just for you."),
-                        this._apiKeyErrorActions
-                    );
-                this._refreshDone(RETRY_RATE_LIMIT_SECONDS);
+                this._apiKeys.shift();
+                this._refresh(verbose);
```

Comment:
This won't work because _updatePending is true and _refresh will return immediately. You can replace it with `this._refreshDone(1);`, to retry in a second.